chainntnfs/neutrinonotify/neutrino.go

```diff
@@ -778,6 +779,21 @@ func (n *NeutrinoNotifier) RegisterSpendNtfn(outpoint *wire.OutPoint,
 			neutrino.EndBlock(&headerfs.BlockStamp{
 				Height: int32(ntfn.HistoricalDispatch.EndHeight),
 			}),
+			neutrino.ProgressHandler(func(processedHeight uint32) {
+				// We persist the rescan progress to achieve incremental
+				// behavior across restarts, otherwise long rescans may
+				// start from the beginning with every restart.
+				if processedHeight > lastProcessedHeight {
+					lastProcessedHeight = processedHeight
+					err := n.spendHintCache.CommitSpendHint(
```

Comment:
Did you check whether it could make sense performance wise to not commit this on every block?

Reply:
IMO this optimisation is not necessary. The wallet rescan also commits every block and we are running this for a long time in mobile clients that have less resources than desktops with no performance issues.
Most of the time actually goes on networking (download the filters).